miners/text/template_miner.py

```diff
@@ -0,0 +1,643 @@
+        """
+        if self.config.miner.compute_remote_gradients:
+            with torch.enable_grad():
+
+                # ---- Set up inputs for gradient computations.
+                inputs_x.requires_grad = True
+                inputs_x = inputs_x.to( self.device )
+                grads_dy = grads_dy.to( self.device )
+                outputs_y = self.nucleus.local_forward( inputs = inputs_x ).to( self.device )
+
+                # ---- The backward call will accumulate gradients on our parameters.
+                if self.config.miner.accumulate_remote_gradients:
+                    torch.autograd.backward (
+                        tensors = [outputs_y],
+                        grad_tensors = [grads_dy]
+                    )
+                    return inputs_x.grad if inputs_x.grad != None else None
+
+                # ---- The backward call will simply compute the gradients without accumulating them.
+                else:
+                    grads_dy = torch.autograd.grad (
+                        outputs = outputs_y,
+                        inputs = inputs_x,
+                        grad_outputs = grads_dy,
+                        only_inputs = True,
+                        create_graph = False,
+                        retain_graph = False
+                    )[0]
+                    return grads_dy
+
+        # if ! compute_remote_gradients, NO-OP.
+        else:
+            return None
+
+    def checkpoint( self ):
+        r""" Optionally Saves, updates and then reloads the miner training state.
+        """
+        last_saved = self.get_saved_state()
+        if last_saved == None or last_saved['epoch_loss'] >= self.epoch_loss:
+            self.save()
+        bittensor.neuron.metagraph.load()
+        bittensor.neuron.metagraph.sync()
+        bittensor.neuron.metagraph.save()
+        #self.reload()
+
+
+    def get_saved_state( self ):
+        r""" Returns a saved state dict or none.
+        """
+        try:
+            return torch.load("{}/model.torch".format( self.config.miner.full_path ))
+        except Exception as e:
+            logger.warning('No saved model found with error: {}', e)
+            logger.info('Initalizing with new model')
+            return None
+
+    def reload( self ):
+        r""" Reloads/updates the training state from the disk.
+        """
+        state_dict = self.get_saved_state()
+
+        # --- loads and syncs metagraph
+        bittensor.neuron.metagraph.load()
+        bittensor.neuron.metagraph.sync()
+        bittensor.neuron.metagraph.save()
+
+        # ---- Load training state.
+        self.epoch = state_dict['epoch']
+        self.epoch_loss = state_dict['epoch_loss']
+        self.global_step = state_dict['global_step']
+        if 'network' in state_dict.keys() and bittensor.neuron.subtensor.network == state_dict['network']: # checks if you are loading into the same network
```

Comment:
I think we should probably just error out here. This should basically never happen.

Reply:
Fixed